### ISSUE TITLE
проверить страницы на адаптивность

### ISSUE THREAD

**Issue by fey** [hexlet-volunteers/hexlet-sicp#1584]
### Problem description

Нет понимания, ок ли наши страницы или есть сломанные

### Proposed solution

сабж

**Comment by JaroslavRusanov** [comment]
Привет! Достаточно ли проверить на эмуляторе DevTools или надо на разном оборудовании?

**Comment by fey** [comment]
@JaroslavRusanov Добрый день.

Девтулс подойдут

**Comment by JaroslavRusanov** [comment]
Готов поработать на новогодних выходных, но я новичок, учусь на frontend на Хекслете (второй проект). Можно задавать вопросы здесь?

**Comment by fey** [comment]
Хм) Тогда лучше задавать вопрсоы в нашем телеграме https://t.me/hexletcommunity/12
Я буду скорее всего отдыхать, но может быть смогу ответить там.

**Comment by JaroslavRusanov** [comment]
Добрый день! Здесь есть ещё одно issue на адаптивность (issue#1448) - это одно и тоже задание?
Также задавал вопрос в волонтерах на ТГ по отступам. Теперь лучше здесь задавать вопросы?

**Comment by fey** [comment]
Добрый день. 1448 скорее про фикс. А тут задача для тестировщиков. 
Можете задавать здесь - тогда контекст задачи будет в одном месте

**Comment by JaroslavRusanov** [comment]
Правильно ли я понимаю, здесь достаточно указать проблему (скинуть скан, или адрес) с описанием где что ломается?

**Comment by fey** [comment]
да, скриншот, адрес страницы. Главное чтобы понять, что сломалось, чтобы разработчики могли это починить.

**Comment by JaroslavRusanov** [comment]
ОК, ловите первую:
[адрес](https://sicp.hexlet.io/ru) при размере меньше 576px отсутствуют горизонтальные отступы. Если у селектора .row убрать margin-left, margin-right отступы возвращаются. При этом изменении на остальных разрешений ничего не ломается на этой странице
![Uploading photo_2024-01-10_15-31-10.jpg…]()

**Comment by JaroslavRusanov** [comment]
[адрес](https://sicp.hexlet.io/ru/chapters) - Оглавление, заголовки (тэги h2, h3) отсутствуют отступы слева на разрешении меньше 576.

**Comment by JaroslavRusanov** [comment]
[адрес](https://sicp.hexlet.io/ru/exercises) и [адрес](https://sicp.hexlet.io/ru/solutions) разрешение меньше 576 также отсутствуют отступы если в селекторе .row и в заголовках отключить margin-left, margin-right отступы появляются

**Comment by JaroslavRusanov** [comment]
Это практически каждый телефон. У меня это Pocophone 5m. Если посмотреть на дэвтулсе - это все Iphone от SE до 14го.

**Comment by JaroslavRusanov** [comment]
Я поясню, что я имел ввиду под разрешением 576px. Это горизонтальное разрешение (по сути width), телефоны редко бывают шире в горизонтали. Не знаю сколько людей будут учиться и решать задачи на телефоне. Теоретически можно повернуть телефон...

**Comment by JaroslavRusanov** [comment]
На примере страницы [осилятора](https://sicp.hexlet.io/ru) (надо зайти авторизованным) в main есть div с классом .container в нем заданы margin-left, margin-right на auto. В них вложены еще div`ы с классом .row в них заданы margin-left, margin-right: calc(var(--bs-gutter-x)*-.5). В девтулсе последние марджи в .row отключаю и появляются нормальные отступы. При этом на остальных width на девтулсе ничего не ломается.

**Comment by JaroslavRusanov** [comment]
@canekg и всем.
[Тут](https://sicp.hexlet.io/ru/solutions) ещё форма сломалась, как мне кажется. (скан старый, ещё без отступов, но сейчас также).
![Uploading 296203508-01fe4b40-6c34-4770-85df-dd5da4a10cc3.jpg…]()
Также на разрешении 768px
![Uploading Безымянный.png…]()

**Comment by lu-lullaby** [comment]
Windows 10, Google Chrome 139.0.7258.67 (64 бит), DewTools, Toggle device toolbar, эмулятор Galaxy Note 3.
Излишне широкий отступ справа.

<img width="1080" height="1928" alt="Image" src="https://github.com/user-attachments/assets/fe70b7c6-012c-4d62-93ff-a3a51e10b76e" />

**Comment by bangiliana** [comment]
В русской и английской версиях отличается размеры отступов между столбцами в таблице Рейтинг: https://sicp.hexlet.io/ratings/top

<img width="1344" height="756" alt="Image" src="https://github.com/user-attachments/assets/91b3c407-ca22-433b-a347-c4e64685f6d9" />



<img width="1344" height="756" alt="Image" src="https://github.com/user-attachments/assets/c4f25241-5acd-4381-9355-7968e55af776" />

**Comment by fey** [comment]
> *This was generated by AI during triage.*

## Состояние: `ready-for-human` — объединяю сюда #1448

В трекере два тикета про одно и то же: этот («проверить страницы на адаптивность», 2023) и #1448 («Fix adaptive for pages», 2022). Второй — скриншот сломанной вёрстки без текста, этот — общая постановка задачи. Оставляю канонической эту, #1448 закрываю как дубликат.

## Почему не `ready-for-agent`

Обе формулировки — «нет понимания, ок ли наши страницы» и картинка — не являются постановкой задачи. Здесь нечего проверить на приёмке: непонятно, какие страницы, какие ширины и что считать сломанным. Агент, взявший такой тикет, будет угадывать.

Задача при этом реальная: три независимых отчёта от разных людей за последний год касались именно вёрстки на узких экранах (#1628 про переполнение комментария, #1668 про поле ввода на мобильных, #1798 про цитату в ответе). То есть проблемы есть, просто они находятся поштучно и вслепую.

## Что предлагаю вместо общей проверки

Разбить на два шага и первый сделать инструментом, а не глазами:

**Шаг 1 — получить список.** Прогнать основные страницы через автоматическую проверку на нескольких ширинах и собрать конкретный перечень: страница, ширина, что именно ломается. Кандидаты в список страниц: главная, оглавление глав, страница главы, список упражнений, страница упражнения с редактором, профиль, рейтинг, журнал активности, логин и регистрация. Ширины: 360, 430, 768, 1024.

Основной измеримый признак — горизонтальная прокрутка body, то есть `scrollWidth > clientWidth`. Он ловит ровно тот класс дефектов, что уже находили вручную, и проверяется без человека.

**Шаг 2 — починить найденное.** По результату завести отдельные тикеты с конкретикой либо один с чеклистом. Вот они уже будут `ready-for-agent`.

## Что нужно от мейнтейнера

1. Подтвердить список страниц и ширин.
2. Решить, делаем ли шаг 1 разово или заводим постоянную проверку в CI. Второе дороже, но иначе список снова протухнет за год.

## Связи

- **#1448** — дубликат, закрыт.
- **#1798** — подтверждённый дефект вёрстки на узких экранах, уже `ready-for-agent`, ждать этот тикет не нужно.
- **#1626**, **#1668** — отправлены репортёрам на перепроверку.

**Comment by fey** [comment]
> *This was generated by AI during triage.*

## Закрываю: аудит адаптивности уехал в план фазы 2 миграции — #1979, пункт 8

Спасибо всем, кто присылал отчёты — @JaroslavRusanov, @lu-lullaby, @RomannRoss, @bangiliana. Работа не выброшена: три года наблюдений из этой ветки перенесены в план как отдельный пункт.

Решение мейнтейнера: не искать сломанную вёрстку на Bootstrap-страницах поштучно сейчас, а проверить адаптивность отдельным проходом во время перехода на Mantine — у него респонсив задаётся своими механизмами (`visibleFrom` / `hiddenFrom`), и чинить старую сетку перед заменой смысла мало.

## Что перенесено в #1979

Конкретика из этой ветки, а не общая постановка:

- **Страницы:** `/ru` (главная у залогиненного), `/ru/chapters`, `/ru/exercises`, `/ru/solutions`, `/ratings/top`
- **Ширины:** 360, 430, 768, 1024
- **Измеримый признак:** `body.scrollWidth > clientWidth` — ловит ровно тот класс дефектов, что находили вручную, и проверяется без человека
- **Куда ложится:** в прогон Playwright из #1981, чтобы список не протух снова за год

## Оговорка

Из перечисленных страниц фаза 1 переносит только `/ru/solutions` (`solution/index`, #1990) — там дефекты уйдут вместе с переносом. Остальные четыре ADR 0004 оставляет на Blade до фазы 2. То есть Mantine сам их не проверит, пункт 8 нужно брать явным тикетом; поэтому он и записан в #1979, а не оставлен в предположениях.

## Связанное

- **#1798** — подтверждённый дефект вёрстки на узких экранах, `ready-for-agent`, ждать миграции не нужно
- **#1626**, **#1668** — закрыты отдельно: первый не воспроизводится, второй уже исправлен
- **#1448** — закрыт ранее как дубликат этого тикета
